File: frontend/src/types/analysis.ts
export type AnalysisStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface TranscriptTurn {
  speaker: string;
  text: string;
  start_seconds?: number | null;
  end_seconds?: number | null;
}

export interface AnalysisResult {
  transcript: TranscriptTurn[];
  summary: string[];
  sentiment: 'positive' | 'neutral' | 'negative' | string;
  sentiment_reason: string;
  confidence: number;
  agent_score?: number | null;
  agent_advice?: string[] | null;
}

export interface JobStatus {
  job_id: string;
  name?: string | null;
  status: AnalysisStatus;
  input_type?: string | null;
  result?: AnalysisResult | null;
  error_message?: string | null;
}

export interface SessionListItem {
  job_id: string;
  name: string | null;
  status: AnalysisStatus;
  input_type: string | null;
  created_at: string;
  sentiment: string | null;
  confidence: number | null;
}

export interface SessionListResponse {
  sessions: SessionListItem[];
  total: number;
  offset: number;
  limit: number;
}

File: worker/app/domain/analysis.py
from dataclasses import dataclass
from enum import StrEnum
from typing import Any


class AnalysisStatus(StrEnum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


@dataclass(frozen=True)
class TranscriptTurn:
    speaker: str
    text: str
    start_seconds: float | None = None
    end_seconds: float | None = None


@dataclass(frozen=True)
class AnalysisOutput:
    transcript: list[dict[str, Any]]
    summary: list[str]
    sentiment: str
    sentiment_reason: str
    confidence: float
